certbot/src/certbot/_internal/client.py

```diff
@@ -397,6 +398,14 @@ def obtain_certificate(self, domains: List[str], old_keypath: Optional[str] = No
             key_size = self.config.rsa_key_size
 
         # Create CSR from names
+        dns_domains = []
+        ip_addrs = []
+        for domain in domains:
+            try:
+                ip_addrs.append(ipaddress.ip_address(domain))
+            except ValueError:
+                dns_domains.append(domain)
```

Comment:
Similar comment to the above: instead of guessing whether a given identifier is an IP address, let's be explicit. `obtain_certificate` should have a named parameter alongside `domains`, called `ip_addresses`. Probably it should have a default value of an empty list.